arch/arm/src/stm32h7/Kconfig

```diff
@@ -284,11 +284,55 @@ config STM32H7_HAVE_SMPS
 	bool
 	default n
 
-config STM32H7_PWR_DIRECT_SMPS_SUPPLY
-	bool "Use direct SMPS supply mode"
+choice
+	prompt "STM32 H7 Power Supply Selection"
+	default STM32H7_PWR_DEFAULT_SUPPLY
 	depends on STM32H7_HAVE_SMPS
+	---help---
+	    The STM32H7x5 and STM32H7x7 support power supply configrations for the VCORE core domain and an external supply,
+	    by configuring the SMPS step-down converter and voltage regulator.
+	    Note:The SMPS step-down converter is not available on all packages.
+
+        Select the power supply mode of your board
+	    - ID:1 LDO supply mode used 'STM32H7_PWR_LDO_SUPPLY' macro
```

Comment:
How does ID:x relate to the choices?

Reply:
This ID is from the data sheet (Reference manual RM0399).
Power control (PWR)
- Figure 22. System supply configurations
![Figure22](https://github.com/apache/nuttx/assets/113963003/18d36af0-652f-4a6b-b032-d0970c4ace03)

- Table 33. Supply configuration control
![Table33](https://github.com/apache/nuttx/assets/113963003/aa00822a-cdfd-4f3b-9601-3c4ac416c8a7)